macros/edr/data_monitoring/monitors_query/column_monitoring_query.sql

```diff
@@ -1,11 +1,18 @@
-{% macro column_monitoring_query(monitored_table, monitored_table_relation, min_bucket_start, max_bucket_end, days_back, column_obj, column_monitors, metric_properties, dimensions) %}
+{% macro column_monitoring_query(monitored_table, monitored_table_relation, min_bucket_start, max_bucket_end, days_back, column_obj, column_metrics, metric_properties, dimensions) %}
     {%- set full_table_name_str = elementary.edr_quote(elementary.relation_to_full_name(monitored_table_relation)) %}
     {%- set timestamp_column = metric_properties.timestamp_column %}
     {% set prefixed_dimensions = [] %}
     {% for dimension_column in dimensions %}
       {% do prefixed_dimensions.append("dimension_" ~ dimension_column) %}
     {% endfor %}
 
+    {% set metric_types = [] %}
+    {% set metric_name_to_type = {} %}
+    {% for metric in column_metrics %}
+        {% do metric_types.append(metric.type) %}
+        {% do metric_name_to_type.update({metric.name: metric.type}) %}
+    {% endfor %}
```

Comment:
Not sure this is the right place, but can we also save `metric_type` in `data_monitoring_metrics`?
(not sure what's the right place for this comment)

Reply:
Hmm the only case where I think it might matter is when the metric definition is **changed** - e.g. the name remains the same but the metric type is changed.
But that makes me think that it should actually be added to `metric_properties`. WDYT?

This achieves two things:
1. The metric will be backfilled if the type changes (same as any other configuration change like `timestamp_column`)
2. In the cloud I implemented logic that only pulls metrics with the most recent metric properties (to ensure we only rely on metrics collected with the most recent configuration - a bit like what happens in `get_anomaly_scores_query`).

Maybe we want to add it only to metrics collected via `collect_metrics` to avoid invalidating all the metrics in existing OSS tests for users (though not sure if it's critical).